ciao-launcher/qemu.go

```diff
@@ -453,19 +472,29 @@ func (q *qemu) startVM(vnicName, ipAddress string) error {
 	}
 
 	if vnicName != "" {
-		net1Param := fmt.Sprintf("nic,model=virtio,macaddr=%s",
-			q.cfg.VnicMAC)
-		var err error
-		var net2Param string
-		net2Param, f, err = computeTapParam(vnicName, q.cfg.NetworkNode)
-		if err != nil {
-			return err
-		}
-		if f != nil {
-			defer f.Close()
+		if q.cfg.NetworkNode {
+			var err error
+			var macvtapParam []string
+			//TODO: @mcastelino get from scheduler/controller
+			numQueues := 4
+			macvtapParam, fds, err = computeMacvtapParam(vnicName, q.cfg.VnicMAC, numQueues)
+			if err != nil {
+				return err
+			}
+			for _, f := range fds {
```

Comment:
Probably more efficient to do
```
defer func(fds []*os.File) {
   for _, f := range fds {
     _ = f.Close()
   }
}(fds)
``` 

This will also keep errcheck happy.  errcheck will complain about defer f.Close().  I realise the code was already like this.  It was on my today list to fix.

Also, I don't see why you check to see if the file handle is nil?  I don't see how that could be the case.